site/topics/functions/functions.rst

```diff
@@ -67,74 +67,139 @@ Calling a Function
-	   x = a_value * 2
-	   y = x/another_value
-	   print(y)
+* If it helps, just think of the parameters as variables that belong to the function
 
-* And we can call it like this::
+* You have already been specifying the parameter in the ``print`` function
+    * The value you want printed out is the value you are setting for the parameter that ``print`` takes
 
-    i_hate_my_boss(10, 4)
+* In reality, you have been using this idea in math class for years
 
-* If this is scaring you, chill
-* Like it or not, you've been doing this for years in math class
-   
-    ``f(x) = x + 5``
-   
-* This is a math *function* that takes a *parameter* 
-* What happens if you say... f(5)
-    * ``f(5) = 5 + 5``
-    * ``f(5) = 10``
+    :math:`f(x) = x * 9/5 + 32`
 
-* IT'S THE SAME WITH THIS HERE!!
-   
-   
-* Let's do one more example with adding two numbers::
+* This is the definition of a function called :math:`f` that takes a parameter :math:`x`
+* If I asked you what :math:`f(20)` is, you can calculate the result
 
-    def add_print(a, b):
-        print(a + b)
+    :math:`f(20) = 20 * 9/5 + 32`
 
-* Now that the function is defined, we can *call* it. Like this:
+    :math:`f(20) = 36 + 32`
 
-    >>> add_print(5, 2)
-    7
+    :math:`f(20) = 68`
 
-* The *call* ``add_print(5, 2)`` gets handled like this:
-    * Python checks to see if it knows about a function named ``add_print``
-        * We just defined ``add_print``, so it does.
-    * When we defined it, we told Python it should have two parameters: ``a`` and ``b``.
-    * Python now takes the values in the call (in this case, ``5`` and ``2``) and assigns those
-     values to the function parameters ``a`` and ``b``.
-        * In other words, the first thing Python does in this case is set ``a = 5`` and ``b = 2``, just like variables. 
-    * Then Python executes the body of the function, with the parameters having their new values.
+* This is the same idea we used in ``celsius_to_fahrenheit``, but in Python instead of our typical math syntax
+
+
+Execution of a Function
+-----------------------
+
+* Below is a simple and arbitrary function (``square_of_sum``) that takes two parameters (``a`` and ``b``) and calculates what the square of their sum is
+
+.. code-block:: python
+    :linenos:
+
+    def square_of_sum(a, b):
+        c = a + b
+        d = c * c
+        return d
+
+
+* If I were to call this function with ``square_of_sum(2, 3)``, Python handles the execution like this
+    #. Python will check to see if it knows about a function called ``square_of_sum``
+    #. Python takes the values supplied to it when called (``2`` and ``3``) and assigns them to their respective parameters
+        * ``a = 2`` and ``b = 3``
+    #. The sum of ``a`` and ``b`` is put into a variable ``c``
+    #. The variable ``c`` is multiplied with itself (effectively squaring it) and the result is assigned to ``d``
+    #. The function returns the value associated with ``d``
```

Comment:
May be worth mentioning the fact that the program flow jumps from the call site to whatever function is called. If `__main__` calls a function, program flow won't return to `__main__` until that function returns a value. This is briefly touched on later, but it seems to fit here.

Reply:
Moving it up is a great idea. I'ma do that.

Reply:
I'll move it into the section in a seperate PR though.